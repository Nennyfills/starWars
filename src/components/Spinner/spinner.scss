.spinner {
    position: relative;
    display: flex;
    justify-content: center;
    align-items: center;
    width: 50px;
    height: 50px;
    margin: auto;
    margin-top: 50px;
}

.spinner-sector {
    border-radius: 50%;
    position: absolute;
    width: 100%;
    height: 100%;
    border: 15px solid transparent;
    mix-blend-mode: overlay;
}

.spinner-text {
    animation: loading-opacity 3s ease-in-out infinite;
    font-size: 2em;
}

.spinner-sector-one {
    animation: rotate 2s ease-out infinite;
    border-top: 15px solid #6dcdec;
}

.spinner-sector-two {
    animation: rotate 2.5s ease-in infinite;
    border-top: 15px solid #9e1daa;
}

.spinner-sector-three {
    animation: rotate 1.5s ease-in-out infinite;
    border-top: 15px solid #884e78;
}

@keyframes rotate {
    from { transform: rotate(0); }
    to { transform: rotate(360deg); }
}

@keyframes loading-opacity {
    0%, 10% {
        opacity: 1;
    }
    25%, 5% {
        opacity: .5;
    }
    1% {
        opacity: .1;
    }
}