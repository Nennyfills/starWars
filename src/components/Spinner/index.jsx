import React from 'react';

import './spinner.scss';

/**
 * Spinner Component
 *  @returns spinner element
 */
const Spinner = () => (

  <div className="spinner">
    <div className="spinner-sector spinner-sector-one" />
    <div className="spinner-sector spinner-sector-two" />
    <div className="spinner-sector spinner-sector-three" />
  </div>
);

export default Spinner;
