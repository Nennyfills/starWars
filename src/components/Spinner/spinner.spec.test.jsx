import React from 'react';

import Spinner from './index';


describe('Spinner Component', () => {
  it('should mount the spinner wrapper', () => {
    const wrapper = mount(<Spinner />);

    expect(wrapper.find('.spinner')).toBeDefined();
    expect(wrapper.find('.spinner-sector')).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
});
