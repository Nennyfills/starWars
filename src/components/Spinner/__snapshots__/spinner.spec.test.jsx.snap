// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Component should mount the spinner wrapper 1`] = `
<Spinner>
  <div
    className="spinner"
  >
    <div
      className="spinner-sector spinner-sector-one"
    />
    <div
      className="spinner-sector spinner-sector-two"
    />
    <div
      className="spinner-sector spinner-sector-three"
    />
  </div>
</Spinner>
`;
